src/zjs_gpio.c

```diff
@@ -421,11 +414,10 @@ static jerry_value_t zjs_gpio_open(const jerry_value_t function_obj,
             zjs_reject_promise(promise_ret, &handle->open_rval, 1);
         }
 
-        jerry_release_value(pinobj);
         return promise_ret;
     }
 
-    return pinobj;
+    return jerry_acquire_value(pinobj);
```

Comment:
Yeah, @martijnthe, that's actually what I intend them to do. Here's an example where I did it because of the error path above. It does result in an extra function call, but worth it for the safety in this kind of case I think.